middlewares/tracing/status_code.go

```diff
@@ -0,0 +1,55 @@
+package tracing
+
+import (
+	"bufio"
+	"net"
+	"net/http"
+)
+
+type statusCodeRecoder interface {
+	http.ResponseWriter
+	GetStatus() int
+}
+
+type statusCodeWithoutCloseNotify struct {
+	http.ResponseWriter
+	status int
+}
+
+// WriteHeader captures the status code for later retrieval.
+func (s *statusCodeWithoutCloseNotify) WriteHeader(status int) {
+	s.status = status
+	s.ResponseWriter.WriteHeader(status)
+}
+
+// GetStatus get response status
+func (s *statusCodeWithoutCloseNotify) GetStatus() int {
+	return s.status
+}
+
+// Hijack hijacks the connection
+func (s *statusCodeWithoutCloseNotify) Hijack() (net.Conn, *bufio.ReadWriter, error) {
+	return s.ResponseWriter.(http.Hijacker).Hijack()
+}
+
+// Flush sends any buffered data to the client.
+func (s *statusCodeWithoutCloseNotify) Flush() {
+	s.ResponseWriter.(http.Flusher).Flush()
```

Comment:
Can you do a type assertion to check whether the underlying response writer is a `Flusher`? `Flush` might actually be called in circumstances where the response writer, or better say the underlying connection, does not have this ability. Something like:

```
if flusher, ok := s.ResponseWriter.(http.Flusher); ok {
    flusher.Flush()
}
```